src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -121,12 +131,63 @@ function useChatUtilsHook(): UseChatUtilsHook {
     });
   };
 
+  const helperSaveChannelDetail = async (channel: ChannelList, response: any) => {
+    if (!localDb) return;
+    const {originalMembers} = getChannelListInfo(response?.channel);
+
+    const promise = originalMembers?.map((member) => {
+      return new Promise((resolve, reject) => {
+        const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+        try {
+          user.saveOrUpdateIfExists(localDb).then(() => resolve(null));
+        } catch (e) {
+          reject(e);
+        }
+      });
+    });
+
+    await Promise.all(promise).catch((e) => console.log('saveChannelDetail error', e));
+  };
+
+  const helperGetChannelDetail = async (channel: ChannelList) => {
+    if (!localDb) return;
+
+    console.log('should fetch channel detail');
+    try {
+      let response;
+      if (channel?.channelType === 'ANON_PM') {
+        response = await AnonymousMessageRepo.getAnonymousChannelDetail('messaging', channel?.id);
+      } else {
+        const channelType = channel?.channelType === 'GROUP' ? 'group' : 'messaging';
+        response = await SignedMessageRepo.getSignedChannelDetail(channelType, channel?.id);
+      }
+      await helperSaveChannelDetail(channel, response);
+    } catch (e) {
+      console.log('getChannelDetail error', e);
+    } finally {
+      setChat((prevChat) => ({
+        ...prevChat,
+        isLoadingFetchingChannelDetail: false
+      }));
+      refresh('channelInfo');
+    }
+  };
+
   const goToChatScreen = (channel: ChannelList, from: AllowedGoToChatScreen) => {
+    setChannelAsRead(channel);
+
+    const isChannelTypeChat: (BetterSocialChannelType | undefined)[] = ['ANON_PM', 'GROUP', 'PM'];
+    const isChat = isChannelTypeChat.includes(channel?.channelType);
+
+    if (isChat) {
+      helperGetChannelDetail(channel);
+    }
+
     setChat({
-      ...chat,
+      isLoadingFetchingChannelDetail: isChat,
       selectedChannel: channel
     });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `goToChatScreen` function is updated to fetch channel details and set the loading state before navigating to the chat screen. This is a good practice as it ensures that the chat screen has all the necessary data before rendering. However, the function now performs asynchronous operations but is not declared as async. Consider declaring this function as async to handle potential errors properly.
```diff
- const goToChatScreen = (channel: ChannelList, from: AllowedGoToChatScreen) => {
+ const goToChatScreen = async (channel: ChannelList, from: AllowedGoToChatScreen) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->